src/lib/optimizer/chunk_statistics.hpp

```diff
@@ -0,0 +1,165 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+#include "storage/base_encoded_column.hpp"
+
+namespace opossum {
+
+class BaseFilter : public std::enable_shared_from_this<BaseFilter> {
+ public:
+  virtual ~BaseFilter() = default;
+
+  virtual bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const = 0;
```

Comment:
Comments for the methods in this file? Especially for `can_prune`, it is not obvious what can be pruned (or not).

Reply:
This is marked as outdated but has not been addressed yet.